src/core/useAtom.ts

```diff
@@ -1,22 +1,6 @@
-import {
-  useCallback,
-  useContext,
-  useDebugValue,
-  useEffect,
-  useReducer,
-} from 'react'
-import type { Reducer } from 'react'
 import type { Atom, Scope, SetAtom, WritableAtom } from './atom'
-import { getScopeContext } from './contexts'
-import { COMMIT_ATOM, READ_ATOM, SUBSCRIBE_ATOM, WRITE_ATOM } from './store'
-import type { VersionObject } from './store'
-
-type ResolveType<T> = T extends Promise<infer V> ? V : T
```

Comment:
Let's keep this. It's a bit awkward, but we don't want to export this internal type (yet).